src/extension.ts

```diff
@@ -147,35 +146,53 @@ class RunLineInTerminal {
     }
 }
 
+const elegantSpinner = require('elegant-spinner');
+
 class RunLineInEditor {
 
     private resultDocument: TextDocument | undefined;
     private parsedResult: object | undefined;
     private queryEnabled = false;
     private query: string | undefined;
     private disposables: Disposable[] = [];
+    private readonly settings: AzureCliToolsSettings = AzureCliToolsSettings.Instance;
+    private runStatusBarItem: StatusBarItem;
+    private interval!: NodeJS.Timer;
+    private spinner = elegantSpinner();
 
     constructor(private status: StatusBarInfo) {
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.toggleLiveQuery', editor => this.toggleQuery(editor)));
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.runLineInEditor', editor => this.run(editor)));
         this.disposables.push(workspace.onDidCloseTextDocument(document => this.close(document)));
         this.disposables.push(workspace.onDidChangeTextDocument(event => this.change(event)));
+
+        this.runStatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left);
     }
 
     private run(source: TextEditor) {
+        var t0 = Date.now();
+        this.interval = setInterval(() => {
+            this.runStatusBarItem.text = `Waiting for response ${this.spinner()}`;
+        }, 50);
+        this.runStatusBarItem.show();
+
         this.parsedResult = undefined;
         this.query = undefined; // TODO
         const cursor = source.selection.active;
         const line = source.document.lineAt(cursor).text;
+        const isPlainText = (line.indexOf('--query') !== -1) || (line.indexOf('-h') !== -1) || (line.indexOf('--help') !== -1);
```

Comment:
Could you do this in a separate PR? And maybe check the output if it starts and ends with `{}` or `[]` to assume it is JSON. That would handle more cases.